lib/guardian/plug/ensure_permissions.ex

```diff
@@ -63,19 +71,38 @@ defmodule Guardian.Plug.EnsurePermissions do
     case Guardian.Plug.claims(conn, key) do
       {:ok, claims} ->
         perms = Map.get(opts, :perms, %{})
-        result = Enum.all?(Map.get(opts, :perm_keys), fn(perm_key) ->
-          found_perms = Guardian.Permissions.from_claims(claims, perm_key)
-          Guardian.Permissions.all?(
-            found_perms,
-            Map.get(perms, perm_key),
-            perm_key
-          )
-        end)
+        result =
```

Comment:
If you have a list of permission sets from the initializer we can just check all the sets. I'm thinking something along the lines of:

```elixir
result = matches_permissions?(conn, key, Map.get(opts, :perm_sets))

if result, do: conn, else: handle_error(conn, opts)
```

```elixir
def matches_permissions?(_, _, []), do: true
def matches_permissions?(conn, key, sets) do
  case Guardian.Plug.claims(conn, key) do
    {:ok, claims} -> 
      Enum.any?(sets, &matches_permission_set?(claims, &1))
    _ -> false
  end
end

defp matches_permission_set?(claims, set) do
  Enum.all?(set, fn({key, required_perms}) -> 
    Guardian.Permissions.from_claims(claims, key)
    |> Guardian.Permissions.all?(required_perms, key)
  end)
end
```